diff --git a/ui/index.ts b/ui/index.ts
index a02a01aa..457905fd 100644
--- a/ui/index.ts
+++ b/ui/index.ts
@@ -32,7 +32,12 @@ export async function attachTabUIToRootElement(): Promise<void> {
 }
 
 export async function attachPopupUIToRootElement(): Promise<void> {
-  const store = await newProxyStore()
+  const storeType = await newProxyStore()
+  // Test 1
+  throw new Error("Forced error")
+
+  // Test 2
+  const store: typeof storeType = await new Promise(() => {})
 
   if (SUPPORT_TABBED_ONBOARDING) {
     const state = store.getState()
