diff --git a/lib/index.js b/lib/index.js
index c32b85b6d78b1f55e9e7951c07530207370e0bd9..dcdde87a2413464d53529f5bddde46e67c0e3db0 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -30,7 +30,12 @@ exports.stringify = function stringify(value, replacer, space, _options) {
 
   var decycled = cycle.decycle(value, options, replacer);
   if (arguments.length === 1) {
-    return JSON.stringify(decycled);
+    return JSON.stringify(decycled, (_, value) => {
+      if (typeof value === "bigint") {
+        return { B_I_G_I_N_T: value.toString() }
+      }
+      return value
+    })
   } else {
     // decycle already handles when replacer is a function.
     return JSON.stringify(decycled, Array.isArray(replacer) ? replacer : null, space);
